edge-util/src/Microsoft.Azure.Devices.Edge.Util/azureLogAnalytics/AzureLogAnalytics.cs

```diff
@@ -0,0 +1,72 @@
+// Copyright (c) Microsoft. All rights reserved.
+namespace Microsoft.Azure.Devices.Edge.Util.AzureLogAnalytics
+{
+    using System;
+    using System.IO;
+    using System.Net;
+    using System.Security.Cryptography;
+    using System.Text;
+
+    /* Sample code from:
+    /* https://github.com/veyalla/MetricsCollector/blob/master/modules/MetricsCollector/AzureLogAnalytics.cs
+    /* https://dejanstojanovic.net/aspnet/2018/february/send-data-to-azure-log-analytics-from-c-code/
+    */
+
+    public class AzureLogAnalytics
+    {
+        public AzureLogAnalytics(string workspaceId, string sharedKey, string logType, string apiVersion = "2016-04-01")
+        {
+            this.WorkspaceId = workspaceId;
+            this.SharedKey = sharedKey;
+            this.LogType = logType;
+            this.ApiVersion = apiVersion;
+        }
+
+        public string WorkspaceId { get; }
+        public string SharedKey { get; }
+        public string ApiVersion { get; }
+        public string LogType { get; }
+
+        public void Post(byte[] content)
+        {
+            string requestUriString = $"https://{this.WorkspaceId}.ods.opinsights.azure.com/api/logs?api-version={this.ApiVersion}";
+            DateTime dateTime = DateTime.UtcNow;
+            string dateString = dateTime.ToString("r");
+            string signature = this.GetSignature("POST", content.Length, "application/json", dateString, "/api/logs");
+            HttpWebRequest request = (HttpWebRequest)WebRequest.Create(requestUriString);
+            request.ContentType = "application/json";
+            request.Method = "POST";
+            request.Headers["Log-Type"] = this.LogType;
+            request.Headers["x-ms-date"] = dateString;
+            request.Headers["Authorization"] = signature;
+            using (Stream requestStreamAsync = request.GetRequestStream())
+            {
+                requestStreamAsync.Write(content, 0, content.Length);
+            }
+
+            using (HttpWebResponse responseAsync = (HttpWebResponse)request.GetResponse())
+            {
+                Console.WriteLine(responseAsync.StatusDescription);
```

Comment:
use Logger instead of console.writeLine

Reply:
There is actually an issue here that I found but decided that I want to address it in the next PR. The controller values actually does not persist between request -- it instantiated a new one every time there is a request; hence, the `&& this.logAnalytics == null ` actually does nothing.  The `AzureLogAnalytics` is always going to be created. 

Since you want to use a logger here, I think it makes sense to deal with the `AzureLogAnalytics` creation now.

Reply:
1. After some investigation, to make the logger persist by itself requires session that does not currently enabled. And it is a lot of hassle to get that working (adding the file dependencies).
2. Since the Azure Log Analytics is its own entity, I don't want to introduce ASP.net logger in here just for the analyzer.

Reply:
I'll work on this issue per discussion in person. 
1. Make `AzureLogAnalytics` a singleton
2. Make `AzureLogAnalytics.post()` takes `logType` as another argument.